File: nest-admin-backend/src/role/dto/create-role.dto.ts
import { ApiProperty } from '@nestjs/swagger'

export class CreateRoleDto {
  @ApiProperty({ example: 'Admin', description: 'The name of the role' })
  roleName: string

  @ApiProperty({ example: 'admin', description: 'The value of the role' })
  roleValue: string

  @ApiProperty({ example: 1, description: 'The status of the role (1: active, 0: disabled)' })
  status?: number

  @ApiProperty({ example: 'Administrator role', description: 'The remark of the role' })
  remark?: string

  @ApiProperty({
    example: ['menu-id-1', 'menu-id-2'],
    description: 'Array of menu IDs to associate with the role',
    required: false,
    type: [String]
  })
  menuIds?: string[]
}

File: java-admin-backend/src/main/java/com/example/admin/dto/CreateRoleDto.java
package com.example.admin.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

/**
 * 角色创建请求DTO
 *
 * @author example
 */
@Schema(description = "角色创建请求参数")
public class CreateRoleDto {

    @NotBlank(message = "角色名称不能为空")
    @Size(max = 50, message = "角色名称长度不能超过50个字符")
    @Schema(description = "角色名称", example = "管理员")
    private String roleName;

    @NotBlank(message = "角色值不能为空")
    @Size(max = 50, message = "角色值长度不能超过50个字符")
    @Schema(description = "角色值", example = "admin")
    private String roleValue;

    @Schema(description = "状态：0-禁用，1-启用", example = "1")
    private Integer status;

    @Size(max = 255, message = "备注长度不能超过255个字符")
    @Schema(description = "备注", example = "系统管理员")
    private String remark;

    // Getters and Setters
    public String getRoleName() {
        return roleName;
    }

    public void setRoleName(String roleName) {
        this.roleName = roleName;
    }

    public String getRoleValue() {
        return roleValue;
    }

    public void setRoleValue(String roleValue) {
        this.roleValue = roleValue;
    }

    public Integer getStatus() {
        return status;
    }

    public void setStatus(Integer status) {
        this.status = status;
    }

    public String getRemark() {
        return remark;
    }

    public void setRemark(String remark) {
        this.remark = remark;
    }
}